aiding.py type hints and auto-generated OpenAPI specs

Sotatek-Patrick-Vu opened the pull request.

Sotatek-Patrick-Vu committed:
aiding.py type hints and auto-generated OpenAPI specs

Sotatek-Patrick-Vu committed:
Generate types for Group Member

Sotatek-Patrick-Vu commented on src/types/keria-api-schema.ts:
```diff
@@ -149,6 +182,345 @@ export interface components {
+            /** @default 0 */
+            nt: string;
+            n?: string[];
+            /** @default 0 */
```
This only is JSDoc annotations, as I know: TypeScript interfaces/types do not support default values—only classes or function parameters do

Sotatek-Patrick-Vu commented on test-integration/randy.test.ts:
```diff
@@ -2,6 +2,7 @@ import { assert, test } from 'vitest';
 import signify from 'signify-ts';
 import { resolveEnvironment } from './utils/resolve-env.ts';
 import { assertOperations, waitOperation } from './utils/test-util.ts';
+import { Tier } from '../src/types/keria-api-schema.ts';
```
I reverted to use `signify.Tier` and remove the importing from `schema.ts`

Sotatek-Patrick-Vu committed:
resolve review comments

Sotatek-Patrick-Vu committed:
fix mock data for aiding test

iFergal commented on src/keri/core/keeping.ts:
```diff
@@ -177,19 +177,25 @@ export class IdentifierManagerFactory {
+
+        if (Algos.salty in aid) {
+            algo = Algos.salty;
+            kargs = aid.salty;
```
Maybe we can refactor this and just return new SaltIdentifierManager here instead? Rather than setting this and using switch later

Sotatek-Patrick-Vu replied:
changed as your suggestion

iFergal commented on test-integration/utils/multisig-utils.ts:
```diff
@@ -85,7 +89,7 @@ export async function acceptMultisigIncept(
     client: SignifyClient,
     groupName: string,
-    aid: HabState,
+    aid: SaltyHabState | GroupHabState | ExternHabState | RandyHabState,
```
I think this can stay as `HabState` since `HabState = SaltyHabState | GroupHabState | ExternHabState | RandyHabState` already

iFergal commented on test/app/test-utils.ts:
Why do we need these changes?

Sotatek-Patrick-Vu replied:
changed to those:
salty: keeper.params() or randy: keeper.params(), ...
just specify property name

iFergal replied:
I still don't understand why the changes are the same. What is the error when using the old (more concise) format?

The only difference between them is the algo field, which is passed using `[algo]`

iFergal commented on test/app/registry.test.ts:
```diff
@@ -67,14 +67,18 @@ describe('registry', () => {
+                prxs: [],
+                nxts: [],
+            },
+        } as unknown) as HabState;
```
Why do we need the `as unknown` casts? Can't we do `const hab: HabState`?

Sotatek-Patrick-Vu replied:
I fixed for this error:
```
Conversion of type '{ prefix: string; state: KeyState; name: string; transferable: true; windexes: never[]; icp_dt: string; randy: { prxs: never[]; nxts: never[]; }; }' to type 'HabState' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type '{ prefix: string; state: KeyState; name: string; transferable: true; windexes: never[]; icp_dt: string; randy: { prxs: never[]; nxts: never[]; }; }' is not comparable to type 'ExternHabState'.
    Type '{ prefix: string; state: { vn?: number[] | undefined; i: string; s: string; p: string; d: string; f: string; dt: string; et: string; kt: string; k: string[]; nt: string; n: string[]; bt: string; b: string[]; c: string[]; ee: { ...; }; di: string; }; ... 4 more ...; randy: { ...; }; }' is missing the following properties from type '{ extern_type: string; pidx: number; }': extern_type, pidxts(2352)
```

Sotatek-Patrick-Vu commented on src/types/keria-api-schema.ts:
```diff
@@ -149,6 +182,345 @@ export interface components {
+            /** @default  */
+            p: string;
+            /** @default 0 */
+            kt: string;
```
fix in keria -> "kt" and "nt" is string|string[]

Sotatek-Patrick-Vu committed:
resolve review comments

Sotatek-Patrick-Vu commented on src/types/keria-api-schema.ts:
removed `?`

Sotatek-Patrick-Vu committed:
resolve review comments

iFergal commented on test/app/test-utils.ts:
```diff
@@ -96,33 +96,124 @@ export async function createMockIdentifierState(
+                di: serder.sad.di ?? '',
+            } as KeyState,
+            icp_dt: '2023-12-01T10:05:25.062609+00:00',
+        } as unknown) as HabState;
```
also these

Sotatek-Patrick-Vu committed:
convert old mockdata + recreate KEL types (all fields are required)

iFergal reviewed (approved):
LGTM, to re-open against WOT

Sotatek-Patrick-Vu committed:
rebase and update code

Sotatek-Patrick-Vu committed:
remove unuse test file

Sotatek-Patrick-Vu committed:
auto generate credential types from KERIA OpenAPI spec (#337)

* spike: Feasibility of generating Signify types from Python type hints via OpenAI

* add and use generated TS types

* demo generated types for credentials().get()

* update types for CredentialState

* resolve review comments

* fix credentials types + add registries's types

* Add more types for credentialing

* resolve review comments

* fix errors/warning for swagger validator

* re-generate types

* re-generate types

* remove debug logs and add multiple overload signatures

* separate generate:types and build

* fix formatting

* use the node.js API instead of starting a child process for generating openAPI types

* remove unuse schemas

* resolve review comment: fix the eslint rules

* update package-lock.json

* update types since keria changed

* update new types from keria

Sotatek-Patrick-Vu committed:
update keria api schema

Sotatek-Patrick-Vu committed:
Merge branch 'main' into feat/aiding_type_hints

Sotatek-Patrick-Vu committed:
fix audit

dependabot[bot] committed:
chore(deps): bump vite from 7.1.6 to 7.1.11 (#349)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 7.1.6 to 7.1.11.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v7.1.11/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 7.1.11
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Sotatek-Patrick-Vu committed:
resolve review comments: improve ts types for tests of aiding

Sotatek-Patrick-Vu committed:
Merge branch 'main' into feat/aiding_type_hints
